src/ecdsa/keygen.rs

```diff
@@ -809,7 +835,11 @@ impl State<KeyGeneratorTraits> for Phase3 {
             log::error!("Phase3 returns errors {:?}", errors);
             return Transition::FinalState(Err(ErrorState::new(errors)));
         }
-        let mut dk = dk.unwrap();
+        let mut dk = dk.expect("invalid paillier decryption key");
+
+        let public_key = from_secp256k1_pk(public_key.expect("cant compute public key"))
+            .expect("invalid full public key");
```

Comment:
I believe, there is already a proper abstraction of `ErrorState` allowing to not use `expect` here at all and propagate errors to the calling code.